### ISSUE TITLE
RedundantSuspendModifier - false positive

### ISSUE THREAD

**Issue by imanushin** [detekt/detekt#2979]
## Expected Behavior
Code below requires ```suspend``` modifier, because code under ```apply``` (which isn't suspend but ```inline```, so it allows suspend code) is non-blocking.

## Observed Behavior
Detekt warns that function below doesn't require ```suspend``` modifier

## Context
```kotlin
suspend fun create(): X {
    return X().apply {
        doSuspend() // suspend method
    }
}
```

## Your Environment
* Version of detekt used: 1.11.0
* Version of Gradle used (if applicable): 6.6
* Operating System and version: MacOS Catalina
* Link to your project (if it's a public repository): private

**Comment by imanushin** [comment]
Root cause - #2961, so was automatically fixed from version 11.1.0.
